File: Alura/estudos/bytebank-herdado/src/SistemaInterno.java

public class SistemaInterno {

	private int senha = 2222;
	
	public void autentica(Autenticavel fa) {
		boolean autenticou = fa.autentica(this.senha);
		if(autenticou) {
			System.out.println("Pode entrar no Sistema.");
		} else {
			System.out.println("Não pode entrar");
		}
	}
	
}

File: Alura/estudos/kotlin/curso_kotlin_007_funcoes/src/main/kotlin/br/com/alura/bytebank/modelo/SistemaInterno.kt
package br.com.alura.bytebank.modelo

class SistemaInterno {

    fun entra(login: Autenticavel, senha: Int, autenticado: () -> Unit = { }) {
        if (login.autentica(senha)) {
            println("Bem vindo ao Bytebank")
            autenticado()
        } else {
            println("Falha na autenticação")
        }
    }

}